httpovergrpc/client/client.go

```diff
@@ -0,0 +1,181 @@
+/*
+ *
+ * Copyright 2019 gRPC authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+
+package httpovergrpc
+
+import (
+	"bufio"
+	"context"
+	"errors"
+	"io"
+	"io/ioutil"
+	"net/http"
+	"net/textproto"
+	"time"
+
+	"google.golang.org/grpc"
+	"google.golang.org/grpc/grpclog"
+
+	pb "google.golang.org/grpc/httpovergrpc/http_over_grpc_proto"
+)
+
+// Transport is an http.Transport that communicates via a HTTPOverGRPC.
+type Transport struct {
+	ctx  context.Context
+	conn *grpc.ClientConn
+	// HTTPOverGRPC client.
+	client pb.HTTPOverGRPCClient
+}
+
+// Close closes the HTTPOverGRPC connection.
+func (t *Transport) Close() error {
+	if t.conn != nil {
+		grpclog.Infof("Closing httpovergrpc connection.")
+		return t.conn.Close()
```

Comment:
I would say we don't even need a `Close` here.  Users are expected to call `Close` on the `ClientConn` for themselves.  Generated protobuf clients don't have a cleanup function.